src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,770 @@
+       Some offset, Some offset)
+    | _ -> None, None
+
+  let bound_texpr d texpr1 = timing_wrap "bounds calculation" (bound_texpr d) texpr1
+end
+
+module D =
+struct
+  include Printable.Std
+  include ConvenienceOps (Mpqf)
+  include VarManagement
+
+  module Bounds = ExpressionBounds
+
+  module Convert = SharedFunctions.Convert (V) (Bounds) (struct let allow_global = true end) (struct let do_overflow_check = false end) (SharedFunctions.Tracked)
+
+  type var = V.t
+
+  let name () = "lin2vareq"
+
+  let to_yojson _ = failwith "ToDo Implement in future"
+
+  let is_bot t = equal t (bot ())
+  let is_bot_env t = t.d = None
+
+  (*this shows "top" for a specific environment to enable the calculations. It is the top_of of all equalities*)
+  let top_of env = {d = Some (EArray.make_empty_array (Environment.size env)); env = env}
+
+  (*Is not expected to be called but implemented for completeness *)
+  let top () = {d = Some (EArray.empty()); env = empty_env}
+
+  (*is_top returns true for top_of array and empty array *)
+  let is_top t = GobOption.exists EArray.is_top_array t.d
+
+  (* prints the current variable equalities with resolved variable names *)
+  let show varM =
+    let lookup i = Var.to_string (Environment.var_of_dim varM.env i) in
+    let show_var i tuple =
+      match tuple with
+      | (None, offset) -> (lookup i) ^ " = " ^ Z.to_string offset ^ ";\n"
+      | (Some index, offset) -> (lookup i) ^ " = " ^ lookup index ^
+                                (if offset <> Z.zero then " + " ^ Z.to_string offset else "") ^ ";\n"
+    in if is_top varM then "⊤\n" else
+      match varM.d with
+      | None -> "⊥\n"
+      | Some arr -> if is_bot varM then "Bot \n" else Array.fold_left (fun acc elem -> acc ^ elem ) "" (Array.mapi show_var arr)
+
+  let pretty () (x:t) = text (show x)
+  let printXml f x = BatPrintf.fprintf f "<value>\n<map>\n<key>\nequalities-array\n</key>\n<value>\n%s</value>\n<key>\nenv\n</key>\n<value>\n%s</value>\n</map>\n</value>\n" (XmlUtil.escape (Format.asprintf "%s" (show x) )) (XmlUtil.escape (Format.asprintf "%a" (Environment.print: Format.formatter -> Environment.t -> unit) (x.env)))
+  let eval_interval ask = Bounds.bound_texpr
+
+  exception Contradiction
+
+  let meet_with_one_conj_with ts i (var, b) =
+    let subst_var ts x (vart, bt) =
+      let adjust (vare, b') =
+        if Option.eq ~eq:Int.equal (Some x) vare then (vart, Z.(b' + bt)) else (vare, b') in
+      BatArray.modify adjust ts in
+    let (var1, b1) = ts.(i) in
+    (match var, var1 with
+     | None, None -> if not @@ Z.equal b b1 then raise Contradiction
+     | None, Some h1 -> subst_var ts h1 (None, Z.(b - b1))
+     | Some j, None -> subst_var ts j (None, Z.(b1 - b))
+     | Some j, Some h1 ->
+       (match ts.(j) with
+        | (None, b2) -> subst_var ts i (None, Z.(b2 + b))
+        | (Some h2, b2) ->
+          if h1 = h2 then
+            (if Z.(b1 <> (b2 + b)) then raise Contradiction)
```

Comment:
Invalid comparison on `Z`.

Reply:
Almost though, it would be correct in `Z.Compare`: https://antoinemine.github.io/Zarith/doc/latest/Z.Compare.html.

Reply:
What exactly? h_x is of type `int` and b_x is of type `Z.t`

Reply:
`b1 <> (b2+b)`

Reply:
which doesn't include `<>` :wink: